### ISSUE TITLE
Fusion 360 - Login no browser opens

### ISSUE THREAD

**Issue by knimix** [bottlesdevs/programs#301]
Hi, today i installed bottles and used the fusion 360 installer / script from bottles.
The problem is when i click on "Sign In" normally a browser should open but this does not happen.
I tried to install a browser inside the bottle (firefox) but this doesn't help.

Is there any fix for this?

**Comment by cjwhitedev** [comment]
I'm having the same problem.

**Comment by shanold** [comment]
Exact same issue.  I even tried installing a browser in the same bottle hoping it would help :(

**Comment by alextrical** [comment]
There was a January update from Autodesk that removed the legacy login method when they Rebranded the product to Fusion. Unfortunately you now need to be using a version of Wine that is at least 8.14 (containing [this patch](https://gitlab.winehq.org/wine/wine/-/merge_requests/3537)) that added the functionality required to login, you will also need to add the mime-type redirection to handle the data given back from the browser once you actually log in.

Once you create a bottle with the correct Version of Wine, run the script below and Change the variable at the top to match the name of bottle you created
```
BOTTLE="Fusion"

#Define a new application link, that we will use for handling the Mimetype. 
flatpak run --command=bottles-cli com.usebottles.bottles add --bottle $BOTTLE -n "AdskIdentityManager" -p "$HOME/.var/app/com.usebottles.bottles/data/bottles/bottles/$BOTTLE/drive_c/Program Files/Autodesk/webdeploy/production/99249ee497b13684a43f5bacd5f1f09974049c6b/Autodesk Identity Manager/AdskIdentityManager.exe" 

#Create mimetype link to handle web login call backs to the Identity Manager
cat > $HOME/.local/share/applications/adskidmgr-opener.desktop << EOL
[Desktop Entry]
Type=Application
Name=adskidmgr Scheme Handler
Exec=flatpak run --command=bottles-cli com.usebottles.bottles run --args-replace -b $BOTTLE -p "AdskIdentityManager" "%u"
StartupNotify=false
MimeType=x-scheme-handler/adskidmgr;
EOL
xdg-mime default adskidmgr-opener.desktop x-scheme-handler/adskidmgr
```

**Comment by cjwhitedev** [comment]
This worked for me, with one caveat. there is one directory listed here that was different for me. its a huge string of numbers and letters containing the Autodesk Identity Manager directory, i had to open my file system and find out what mine was named, replaced it in the command, and everything worked fine. thanks!

**Comment by alextrical** [comment]
hmm, its possible that Autodesk have released yet another update, and as a result the string of numbers has changed. Once i can get the "Run_script" action working in the manifest, i will look into detecting the latest folder from the registry records to allow it to be dynamic

**Comment by knimix** [comment]
@alextrical  yes thanks that worked, but some things were easier to do via the gui than via commands, especially with the long paths which are always different

**Comment by alextrical** [comment]
Ideally we want all of the steps required added into the application manifest removing the need for the end user to run scripts

**Comment by LightningPORTO** [comment]
I'm still having trouble loggin in. 

I followed the steps above with the correct path and it did open the login page on the browser.
After the browser opens and I input the correct login data, it attempts to redirect me to the product but the F360 is still in the sign in page.

**Comment by jaype87** [comment]
Unfortunately, this doesn't work for me.

Browser opens, but after logging in, nothing happens in fusion.
Manually running xdg-open with the url shows `Program "AdskIdentityManager" not found`
So I removed the quotes around AdskIdentityManager from the Exec command and now it is able to execute without complaining. However, the bottles UI crashes and Fusion still waits for conformation.

Any further ideas?

**Comment by Thermionix** [comment]
[Dynamically detecting AdskIdentityManager](https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/issues/391#issuecomment-2069042232)
a draft outline for something in [fusion.yml](https://github.com/bottlesdevs/programs/blob/main/Software/fusion.yml)
````
- action: add-adskidmgr   # - can we call this as part of setup? can we interact with main user $HOME ?
  script: |
    cat > $HOME/.local/share/applications/adskidmgr-opener.desktop << EOL
[Desktop Entry]
Type=Application
Name=adskidmgr Scheme Handler
Exec=flatpak run --command=bottles-cli com.usebottles.bottles run --args-replace -b !bottle_name -p "AdskIdentityManager" "%u"
StartupNotify=false
MimeType=x-scheme-handler/adskidmgr;
EOL
    xdg-mime default adskidmgr-opener.desktop x-scheme-handler/adskidmgr

- action: AdskIdentityManager
  script: |
      $(find "!bottle_drive" -name "AdskIdentityManager.exe" | head -1 | xargs -I '{}' echo {})

````

Also, will really need to be able to define (unreleased) [wine > 9.8](https://gitlab.winehq.org/wine/wine/-/merge_requests/5525) for latest [installer to succeed](https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/issues/408)

**Comment by Ryan3141** [comment]
I found that the .desktop for adskidmgr-opener had an issue with the path to AdskIdentityManager.exe, causing the link from the browser to never allow fusion to login.  This is likely due to the backslashes somewhere in one of the script, but I don't know where to look.  The original arguements were:
'C:\rogram Files\utodesk\ebdeploy\roduction\738a7f16475c27414ba957f7f0b2ee5b34e4e6d\utodesk Identity Manager\dskIdentityManager.exe' %u
Until I corrected it to:
'C:\Program Files\Autodesk\webdeploy\production\b738a7f16475c27414ba957f7f0b2ee5b34e4e6d\Autodesk Identity Manager\AdskIdentityManager.exe' %u

Note: I am using Manjaro, not sure if that affects things.

**Comment by DalSheron** [comment]
> [Dynamically detecting AdskIdentityManager](https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/issues/391#issuecomment-2069042232) a draft outline for something in [fusion.yml](https://github.com/bottlesdevs/programs/blob/main/Software/fusion.yml)
> 
> ```
> - action: add-adskidmgr   # - can we call this as part of setup? can we interact with main user $HOME ?
>   script: |
>     cat > $HOME/.local/share/applications/adskidmgr-opener.desktop << EOL
> [Desktop Entry]
> Type=Application
> Name=adskidmgr Scheme Handler
> Exec=flatpak run --command=bottles-cli com.usebottles.bottles run --args-replace -b !bottle_name -p "AdskIdentityManager" "%u"
> StartupNotify=false
> MimeType=x-scheme-handler/adskidmgr;
> EOL
>     xdg-mime default adskidmgr-opener.desktop x-scheme-handler/adskidmgr
> 
> - action: AdskIdentityManager
>   script: |
>       $(find "!bottle_drive" -name "AdskIdentityManager.exe" | head -1 | xargs -I '{}' echo {})
> ```
> 
> Also, will really need to be able to define (unreleased) [wine > 9.8](https://gitlab.winehq.org/wine/wine/-/merge_requests/5525) for latest [installer to succeed](https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/issues/408)

doing that stuff from the linked forum post leaves me no closer to getting Fusion 360 to fully load, I wonder what else there is to do.
